conans/model/profile.py

```diff
@@ -87,8 +55,66 @@ def update_settings(self, new_settings):
             res.update(new_settings)
             self.settings = res
 
-    def update_package_settings(self, package_settings):
+
+PackageSettings = _ProfileSettings
+
+
+class Profile(_ProfileSettings):
+    """A profile contains a set of setting (with values), environment variables
+    """
+
+    def __init__(self):
+        # Sections
+        super(Profile, self).__init__()
+        self._package_settings = defaultdict(PackageSettings)
+        self.env_values = EnvValues()
+        self.options = OptionsValues()
+        self.build_requires = OrderedDict()
+
+    def package_settings(self, package_name):
+        return self._package_settings[package_name]
+
+    def process_settings(self, cache, preprocess=True):
+        super(Profile, self).process_settings(cache, preprocess)
+
+        for package_name, settings in self._package_settings.items():
+            try:
+                settings.process_settings(cache, preprocess=preprocess)
+            except ConanException as e:
+                raise ConanException("[{}] {}".format(package_name, e))
+
+    def dumps(self):
+        result = ["[settings]", ]
+        result.append(self._dumps(None))
+        for package, values in self._package_settings.items():
+            result.append(values._dumps(package))
+
+        result.append("[options]")
+        result.append(self.options.dumps())
```

Comment:
why options aren't using pattern while build_requires do? they have the same structure/syntrax